aurora/src/main/java/src/config/ChannelManager.java

```diff
@@ -0,0 +1,129 @@
+package src.config;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.market.banica.common.channel.ChannelRPCConfig;
+import com.market.banica.common.util.ApplicationDirectoryUtil;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import src.model.ChannelProperty;
+
+import javax.annotation.PreDestroy;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.AbstractMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@Component
+public class ChannelManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ChannelManager.class);
+
+    private final Map<String, ManagedChannel> channels = new ConcurrentHashMap<>();
+
+
+    @Autowired
+    public ChannelManager() {
+        this.populateChannels(this.readChannelsConfigsFromFile());
+        LOGGER.debug("Map size after population {}", channels.size());
+    }
+
+
+    public Optional<ManagedChannel> getChannelByKey(String key) {
+        LOGGER.debug("Getting channel with key {}", key);
+        return Optional.ofNullable(channels.get(key));
+    }
+
+    protected void addChannel(String key, ChannelProperty value) {
+        LOGGER.info("Adding new channel {} to map", key);
+        Map.Entry<String, ManagedChannel> entry = this.convertPropertyToChannel(new AbstractMap.SimpleEntry<>(key, value));
+
+        this.channels.put(entry.getKey(), entry.getValue());
+    }
+
+    protected void deleteChannel(String key) {
+        LOGGER.info("Deleting channel {} to map", key);
+        ManagedChannel managedChannel = this.channels.remove(key);
+        this.shutDownChannel(managedChannel);
+    }
+
+    protected void editChannel(String key, ChannelProperty value) {
+        LOGGER.info("Editing channel {} to map", key);
+        Map.Entry<String, ManagedChannel> entry = this.convertPropertyToChannel(new AbstractMap.SimpleEntry<>(key, value));
+
+        ManagedChannel channel = this.channels.remove(key);
+
```

Comment:
what if there is no channel with this key, add null check

Reply:
fixed

Reply:
it would be better if the checks are inside channel manager, so it is the only one who takes care of the channels